checks.d/php_fpm.py

```diff
@@ -0,0 +1,120 @@
+# 3p
+import requests
+
+# project
+from checks import AgentCheck
+from util import headers
+
+
+class PHPFPMCheck(AgentCheck):
+    """
+    Tracks basic php-fpm metrics via the status module
+    Requires php-fpm pools to have the status option.
+    See http://www.php.net/manual/de/install.fpm.configuration.php#pm.status-path for more details
+    """
+
+    SERVICE_CHECK_NAME = 'php_fpm.alive'
+
+
+    GAUGES = {
+        'listen queue': 'php_fpm.listen_queue',
+        'max listen queue': 'php_fpm.max_listen_queue',
+        'listen queue len': 'php_fpm.listen_queue_len',
+        'idle processes': 'php_fpm.idle_processes',
+        'active processes': 'php_fpm.active_processes',
+        'total processes': 'php_fpm.total_processes',
+        'max active processes': 'php_fpm.max_active_processes',
+        'max children reached': 'php_fpm.max_children_reached'
+    }
+    COUNTERS = {
+        'accepted conn': 'php_fpm.accepted_conn',
+        'slow requests': 'php_fpm.slow_requests'
+    }
+
+    def check(self, instance):
+        status_url = instance.get('status_url')
+        ping_url = instance.get('ping_url')
+
+        auth = None
+        user = instance.get('user')
+        password = instance.get('password')
+
+        tags = instance.get('tags', [])
+
+        if user and password:
+            auth = (user, password)
+
+        if status_url is None and ping_url is None:
+            raise Exception("No status_url or ping_url specified for this instance")
+
+        pool = None
+        status_exception = None
+        if status_url is not None:
+            try:
+                pool = self._process_status(status_url, auth, tags)
+            except Exception as e:
+                status_exception = e
+                pass
+
+        if ping_url is not None:
+            self._process_ping(ping_url, auth, tags, pool)
+
+        if status_exception is not None:
```

Comment:
I should have shared that in a comment here, we discussed this with @remh .
The problem of running `ping` first is that we cannot use the pool name parsed from `status` to tag the service check which is :disappointed:
And at the same time we don't want `ping` to be skipped because `status` raised, they're indeed two different endpoints set up in php fpm.